scripts/utils.js

```diff
@@ -46,4 +45,125 @@ searchBar.addEventListener('keyup', (event) => {
     else if(event.code === "Backspace" || event.code === "Delete") {
         document.querySelector("main").innerHTML = mainContent.replace(removeMatch, "");
     }
-});
\ No newline at end of file
+});
+
+/*.................................................................................................................*/
+
+// Table Header creation
+var tableHeader = (httpMethod, tableHeaderUrl) => {
+    createJsonRequest(httpMethod, tableHeaderUrl, function(err, response) {
+
+        const table = document.querySelector("#pageTable thead");
+        const tableHeadRow = document.createElement("tr");
+
+        for(let column of response) {
+
+            const heading = document.createElement("th");
+            heading.innerText = column.name;
+            heading.style.backgroundColor = "#6aa1b2";
+
+            //sortable: true --> if true make a sort button, align text to center to match button
+            if (column.sortable) {
+                const button = document.createElement("button");
+                button.innerText = "Sort";
+                heading.style.textAlign = "center";
+                heading.appendChild(button);
+            }
+            tableHeadRow.appendChild(heading);
+        }
+        table.appendChild(tableHeadRow)
+        
+    });
+}
+export {tableHeader};
+
+/*.................................................................................................................*/
+
+// Table Content creation
+var tableContent = (httpMethod, tableHeaderUrl, tableContentUrl) => {
+
+    createJsonRequest(httpMethod, tableHeaderUrl, function(err, tableColumns) {
+
+        let keys = [], types = [],texts =[];
+        for(let column of tableColumns) {
+            keys.push(column.key);
+            types.push(column.type);
+            texts.push(column.text);
+        }
+        
+        createJsonRequest(httpMethod, tableContentUrl, function(err, tableRows) {
+
+            const table = document.querySelector("#pageTable tbody");
+            console.log(tableRows)
+
+            for(let column of tableRows) {
```

Comment:
Better if you could provide some comments against each iterations - what is does actually

Reply:
Moved table function from utils.js to to table.js and added appropriate comments in the latest commit.